java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestGenerationUtils.java

```diff
@@ -358,6 +354,42 @@ private static String getSuggestedMethodNameByAnnotation(String annotation) {
         }
     }
 
+    /**
+     * return modifier bit mask.
+     */
+    private static int getTestMethodModifiers(ITypeBinding typeBinding, TestKind kind,
+            String annotation, String methodName) {
+        int modifiers = Modifier.NONE;
+        // JUnit 4's test method must be public
+        if (kind == TestKind.JUnit) {
+            modifiers |= Modifier.PUBLIC;
+        }
+
+        // @BeforeClass and @AfterClass in JUnit 4 & 5 needs static modifier
+        if (needStaticModifier(kind, annotation)) {
+            modifiers |= Modifier.STATIC;
+        }
+
+        ITypeBinding superClass = typeBinding.getSuperclass();
+        while (superClass != null) {
+            final IMethodBinding[] superMethods = superClass.getDeclaredMethods();
+            for (final IMethodBinding superMethod : superMethods) {
+                if (Objects.equals(superMethod.getName(), methodName)) {
+                    final int superModifiers = superMethod.getModifiers();
+                    // Cannot reduce the visibility of the inherited method
+                    if (Modifier.isPublic(superModifiers)) {
```

Comment:
both `protected`, `public` methods in super class are visible to subclass, and could override it in subclass.